### PR TITLE
Create rough implementation of functionality (not tested)

### PR DESCRIPTION
**Pull request by MitchPetersen**
Created rough implementation using the new API re-write. Needs testing/review

### CODE REVIEW

File: TransactionService/docker_context/TransactionServiceApi.py [hunk 1]
```diff
@@ -10,81 +9,92 @@
...
 
 @app.route(endpoints.quote_endpoint, methods=['GET'])
-async def getQuote(request, trans_num, user_id, stock_symbol):
-    data = {
-        'transaction_num': trans_num,
-        'user_id': user_id,
-        'stock_symbol': stock_symbol
-    }
+async def getQuote(request):
+    data = request.json
```
Comment (MitchellGRead) [inline TransactionService/docker_context/TransactionServiceApi.py]: you're going to have to revert this change back along with the getQuote method signature. This is a GET HTTP request and won't have a reqest.json body

File: TransactionService/docker_context/TransactionServiceApi.py [hunk 2]
```diff
@@ -10,81 +9,92 @@
...
 
 
 # BUY ENDPOINTS ------------------------------------------------
 @app.route(endpoints.buy_endpoint, methods=['POST'])
 async def buyStock(request):
-    res, err = validateRequest(request.json, buy_schema)
+    data = request.json
+    res, err = validateRequest(data, buy_schema)
     if not res:
-        return response.json(errorResult(err, request.json), status=400)
```
Comment (MitchellGRead) [inline TransactionService/docker_context/TransactionServiceApi.py:33]: We passed in the request.json object here in case cause the validation checks to make sure the json object exists in the request

File: TransactionService/docker_context/apiListeners.py [hunk 3]
```diff
@@ -20,6 +26,42 @@ async def connectRedis(app, loop):
     )
 
 
+async def initRedisHandler(app, loop):
+    app.config['redisHandler'] = RedisHandler(app.config['redisPool'], app.config['client'], config.DATABASE_SERVER_IP,
```
Comment (MitchellGRead) [inline TransactionService/docker_context/apiListeners.py]: would we want the redis handler to be the one that manages the connection pool instead of the service?

File: TransactionService/docker_context/apiListeners.py [hunk 4]
```diff
@@ -20,6 +26,42 @@ async def connectRedis(app, loop):
...
+        config.AUDIT_SERVER_IP,
+        config.AUDIT_SERVER_PORT
+    )
+
+
+async def initLegacyStock(app, loop):
+    # stockHost = "192.168.4.2"
+    stockHost = 'dummy-stock-1'
+
+    stockPort = 4444
```
Comment (MitchellGRead) [inline TransactionService/docker_context/apiListeners.py]: we should add the quote server port and ip to the config file

File: TransactionService/docker_context/handlers/BaseLogicHandler.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+class BaseLogicHandler:
+
+    def __init__(self, LegacyStockServerHandler, redis, RedisHandler, audit, client, ip, port):
+        self.LegacyStockServer = LegacyStockServerHandler
+        self.redis = redis
+        self.RedisHandler = RedisHandler
+        self.audit = audit
+        self.client = client
+        self.dbmURL = f'HTTP://{ip}:{port}'
```
Comment (MitchellGRead) [inline TransactionService/docker_context/handlers/BaseLogicHandler.py]: http should be lowercase

File: TransactionService/docker_context/handlers/BaseLogicHandler.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    async def commitBuy(self, trans_num, user_id):
+        if self.redis.exists(user_id + "_BUY"):
+            buyObj = pickle.loads(self.redis.get(user_id + "_BUY"))
+            now = datetime.now()
+            timeDiff = (now - buyObj["time"]).total_seconds()
+            if timeDiff <= 60:
+                data = {"user_id": user_id, "funds": buyObj["amount"],
+                        "stock_symbol": buyObj["stock_id"], "stock_amount": buyObj["amount_of_stock"]}
+                result = await self.postRequest(self.dbmURL+'/stocks/buy_stocks', data)
+                self.redis.delete(user_id + "_BUY")
```
Comment (MitchellGRead) [inline TransactionService/docker_context/handlers/BaseLogicHandler.py]: this is what I meant before with the redis connection pool being in the redis handler. As it is we have redis operations being done in different areas. What do you think?

File: TransactionService/docker_context/handlers/LegacyStockServerHandler.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        self.ip = ip
+        self.port = port
+        self.redis = redis
+        self.audit = audit
+
+    def quoteSocket(self):
+        stockSocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        print("Attempting Connection to Quote Server")
+        while stockSocket.connect_ex((self.ip, self.port)) != 0:
+            sleep(1)
```
Comment (MitchellGRead) [inline TransactionService/docker_context/handlers/LegacyStockServerHandler.py]: could this potentially hang execution of the service, therefore, the user if we were unable to connect? could implement a timeout for it

File: TransactionService/docker_context/handlers/BaseLogicHandler.py [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+    async def addFunds(self, trans_num, user_id, amount):
+        data = {'user_id': user_id, 'funds': float(amount)}
+        result = await self.postRequest(self.dbmURL+"/funds/add_funds", data)
+        return result
+
+    async def getQuote(self, trans_num, user_id, stock_id):
+        result = await self.LegacyStockServer.getQuote(trans_num, user_id, stock_id)
+        return result
+
+    async def BuyStock(self, trans_num, user_id, stock_symbol, amount):
```
Comment (MitchellGRead) [inline TransactionService/docker_context/handlers/BaseLogicHandler.py]: camel casing for the method

File: TransactionService/docker_context/handlers/BaseLogicHandler.py [hunk 9]
```diff
@@ -0,0 +1,117 @@
+import pickle
+from datetime import datetime
+
+class BaseLogicHandler:
```
Comment (MitchellGRead) [inline TransactionService/docker_context/handlers/BaseLogicHandler.py]: just thinking we rename this to TransactionHandler along with BaseLogic --> TransactionLogic

**Review by MitchellGRead — CHANGES_REQUESTED** [review]

File: TransactionService/docker_context/TransactionServiceApi.py (hunk 1 above)
Comment (MitchPetersen) [reply TransactionService/docker_context/TransactionServiceApi.py]: I believe this is fixed. I wasn't putting 2 and 2 together of why everything was being passed like this. This makes sense.

File: TransactionService/docker_context/TransactionServiceApi.py (hunk 2 above)
Comment (MitchPetersen) [reply TransactionService/docker_context/TransactionServiceApi.py:33]: reverted back to checking request.json for all except getQuote.

File: TransactionService/docker_context/apiListeners.py (hunk 3 above)
Comment (MitchPetersen) [reply TransactionService/docker_context/apiListeners.py:30]: I intended the redis handler to be more of a data handler, But I can combine these if you think it'd be better.

File: TransactionService/docker_context/apiListeners.py (hunk 4 above)
Comment (MitchPetersen) [reply TransactionService/docker_context/apiListeners.py]: Added and changed the reference here

File: TransactionService/docker_context/handlers/BaseLogicHandler.py (hunk 9 above)
Comment (MitchPetersen) [reply TransactionService/docker_context/handlers/BaseLogicHandler.py]: Rename complete.

File: TransactionService/docker_context/handlers/BaseLogicHandler.py (hunk 6 above)
Comment (MitchPetersen) [reply TransactionService/docker_context/handlers/BaseLogicHandler.py]: We can talk about this in the meeting in a bit, and I can rewrite if necessary.

File: TransactionService/docker_context/handlers/LegacyStockServerHandler.py (hunk 7 above)
Comment (MitchPetersen) [reply TransactionService/docker_context/handlers/LegacyStockServerHandler.py:19]: You are right that this could hang the operation, but only if the legacy stock server is down, in which case there are likely bigger issues. I can include a more proper timeout, but it would likely just end up forcing an error back through to the web server saying that we couldn't establish a connection to the quote server.

**Review by MitchPetersen — COMMENTED** [review]
Fixed where needed, pending discussion on redis and timeout for socket.
